contracts/base/BaseAccount.sol

```diff
@@ -82,14 +82,13 @@ contract BaseAccount is IBaseAccount {
         address entryPointAddress = _ENTRYPOINT;
         /// @solidity memory-safe-assembly
         assembly {
-            mstore(0x14, to) // Store the `to` argument.
-            mstore(0x34, amount) // Store the `amount` argument.
-            mstore(0x00, 0x205c2878000000000000000000000000) // `withdrawTo(address,uint256)`.
+            mstore(0x14, to) // Store the `to` argument directly at memory location 0x14.
+            mstore(0x34, amount) // Store the `amount` argument at memory location 0x34.
+            mstore(0x00, 0x205c2878000000000000000000000000) // Store the function selector at memory location 0x00.
             if iszero(call(gas(), entryPointAddress, 0, 0x10, 0x44, codesize(), 0x00)) {
-                returndatacopy(mload(0x40), 0x00, returndatasize())
-                revert(mload(0x40), returndatasize())
+                returndatacopy(0x0, 0x00, returndatasize()) // Copy return data to 0x0.
+                revert(0x0, returndatasize()) // Revert with data at 0x0.
             }
-            mstore(0x34, 0) // Restore the part of the free memory pointer that was overwritten.
```

Comment:
why was this removed? if the call succeeds you still need to reset this from the `mstore(0x34, amount)` above.

Reply:
@MrToph can you please review this again?

Reply:
it is now added back again